microsoft/testsuites/vm_extensions/azureperformancediagnostics.py

```diff
@@ -0,0 +1,192 @@
+import string
+from datetime import datetime, timezone
+from typing import Any
+
+from assertpy import assert_that
+
+from lisa import (
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.sut_orchestrator.azure.common import (
+    check_or_create_storage_account,
+    get_or_create_storage_container,
+    get_storage_credential,
+)
+from lisa.sut_orchestrator.azure.features import AzureExtension
+from lisa.sut_orchestrator.azure.platform_ import AzurePlatform
+from lisa.testsuite import TestResult
+from lisa.util import SkippedException, generate_random_chars
+
+
+@TestSuiteMetadata(
+    area="vm_extension",
+    category="functional",
+    description="Tests for the Azure Performance Diagnostics VM Extension",
+)
+class AzurePerformanceDiagnostics(TestSuite):
+    def before_case(self, log: Logger, **kwargs: Any) -> None:
+        node = kwargs["node"]
+        if not self._is_supported_linux_distro(node):
+            raise SkippedException(
+                f"{str(node.os.information.full_version)} is not supported."
+            )
+
+    @TestCaseMetadata(
+        description="""
+        Installs and runs the Azure Performance Diagnostics VM Extension.
+        Verifies a report was created and uploaded to the storage account.
+        Deletes the VM Extension.
+        """,
+        priority=3,
```

Comment:
How long is this case taken? You can have a P1 test case to be covered in most test scenarios.